### PR TITLE
Add cache purges for updates, deletes

### PR DESCRIPTION
**Pull request by rowanbeentje**
I've had this sitting around for ages, kept meaning to write tests.  I think as we discussed none of the api stuff is currently tested, which I thought was fine – I could set up some basic mocks, etc – but there's not even a setup to mock `fetch()`es on this project so I've given up on that front :)

Tried to follow the rest of the file's es6 approach - yes to template strings and promises, no to arrow functions?

### CODE REVIEW

**Comment by matthew-andrews** [comment]
arrow functions are totally allowed now

File: server/lib/api-v3.js [hunk 1]
```diff
@@ -51,11 +55,37 @@ function update(id, opts) {
...
+		})
+		.then(function(res) {
+			if (res.ok) {
+				if (opts.verbose) {
+					logger.info(`event=PURGED uuid=${id}`);
+				}
+			} else {
+				logger.info(`event=PURGE_FAILED uuid=${id}`);
+			}
+			return !!res.ok;
```
Comment (matthew-andrews) [inline server/lib/api-v3.js]: feel [oddly qualified](https://github.com/whatwg/fetch/issues/18#issuecomment-71345650) to say that `.ok` is a boolean so `!!` is unnecessary 🙃

**Comment by matthew-andrews** [comment]
one stylistic comment but really it's good to go.  merge at will.  💚

**Comment by rowanbeentje** [comment]
I'll leave it for someone else to merge next week as now's probably the wrong time :D

**Pull request merged**